woocommerce/woocommerce-gutenberg-products-block
# Show validation errors again

Fixes #2112.
Closes #2096.

This PR makes it so validation errors show again in the _Checkout_ block, it also makes validation errors added by extensions to be displayed at the same time default validation errors are shown.

### Screenshots
<img src="https://user-images.githubusercontent.com/3616980/78539510-f9811a00-77f2-11ea-87ad-a77ba3c6ac67.png" width="466" />

### How to test the changes in this Pull Request:

In order to test it, let's force the payment method to return an error in certain circumstances. In `payment-method-extensions/payment-methods/stripe/credit-card/use-checkout-subscriptions.js` [L111](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/payment-method-extensions/payment-methods/stripe/credit-card/use-checkout-subscriptions.js#L111), paste something like this:
```JS
if ( billingData.postcode !== 'ABCD' ) {
	paymentStatus
		.setPaymentStatus()
		.error( 'Stripe found an error with your postcode.' );
	const key = billingData.shippingAsBilling
		? 'shipping-postcode'
		: 'billing-postcode';
	return {
		[ key ]: {
			message: 'Postcode must be ABCD.',
		},
	};
}
```
1. Go to the _Checkout_ block and click on _Place Order_ when you have some validation errors (and the postal code is different from `ABCD`).
2. Make sure the validation errors + the payment method errors appear all at once.
3. Make sure there were no API calls before all validation errors were fixed.
